tiktoken-rs/src/api.rs

```diff
@@ -328,3 +306,70 @@ mod tests {
         assert!(max_tokens > 0);
     }
 }
+
+#[cfg(feature = "async-openai")]
+pub mod async_openai_api {
+    use anyhow::Result;
+    use async_openai::types::ChatCompletionRequestMessage;
+
+    pub fn async_openai_num_tokens_from_messages(
+        model: &str,
+        messages: &[ChatCompletionRequestMessage],
+    ) -> Result<usize> {
+        let messages = messages
+            .iter()
+            .map(|m| super::ChatCompletionRequestMessage {
+                role: m.role.to_string(),
+                name: m.name.clone(),
+                content: m.content.clone(),
+            })
+            .collect::<Vec<_>>();
```

Comment:
Consider implementing a from trait from ChatCompletionRequestMessage into ChatCompletionRequestMessage, then the implementation becomes
```suggestion
        let messages = messages
            .iter()
            .map(|m| m.into())
            .collect::<Vec<_>>();
```

Reply:
Consider
```

impl From<super::ChatCompletionRequestMessage> for ChatCompletionRequestMessage {
    fn from(m: super::ChatCompletionRequestMessage) -> Self {
        Self {
                role: m.role.to_string(),
                name: m.name.clone(),
                content: m.content.clone(),
        }
    }
}
```